### PR TITLE
error-reporting: Address doc inconsistencies

### PR DESCRIPTION
**Pull request by DominicKramer**
In the README file:

* Removed documentation on automatically handling
  uncaughtExceptions since that logic no longer exists in the
  code.
* Removed the statement saying that this project is not an
  official Google project.
* Added clarity to what configuration is needed in different
  environments.
* Removed an invalid reference to the Stackdriver Trace agent
  for Node.js.
* Did some refactoring to make the documentation easier to read.

### CODE REVIEW

File: packages/error-reporting/README.md [hunk 1]
```diff
@@ -16,11 +16,12 @@ applications running in almost any environment. Here's an introductory video:
 ## Prerequisites
 
 1. Your application needs to use Node.js version 4.x or greater.
-1. You need a [Google Cloud project](https://console.cloud.google.com). Your application can run anywhere, but errors are reported to a particular project.
-1. [Enable the Stackdriver Error Reporting API](https://console.cloud.google.com/apis/api/clouderrorreporting.googleapis.com/overview) for your project.
-1. The module will only send errors when the `NODE_ENV` environment variable is
+2. You need a [Google Cloud project](https://console.cloud.google.com). Your application can run anywhere, but errors are reported to a particular project.
```
Comment (ofrobots) [inline packages/error-reporting/README.md]: Keep the 1's to allow easier updates to the markdown in the future. This is valid (and preferred) way to do numbered lists in markdown. Elsewhere in the file too.

File: packages/error-reporting/README.md [hunk 2]
```diff
@@ -110,7 +115,7 @@ When running on Google Cloud Platform, we handle these for you automatically.
 
 ## Configuration
 
-The following code snippet lists all available configuration options. All configuration options are optional.
+The following code snippet lists all available configuration options. Except for the `keyFilename` and `credentials` options, all configuration options are optional.  Whether or not the `keyFilename` and `credentials` options are required is outlined in the [Running on Google Cloud Platform](#running-on-google-cloud-platform) section above.
```
Comment (ofrobots) [inline packages/error-reporting/README.md]: Aren't `keyFilename` and `credentials` optional too, specially on GCP? What am I missing?

File: packages/error-reporting/README.md (hunk 1 above)
Comment (DominicKramer) [inline packages/error-reporting/README.md]: Done.

File: packages/error-reporting/README.md (hunk 2 above)
Comment (DominicKramer) [inline packages/error-reporting/README.md]: I've updated the document to state that all of the config options are optional.

**Pull request merged**